meson_init.sh

```diff
@@ -59,16 +78,15 @@ if [[ ! -n "$(command -v ninja)" ]]; then
 fi
 
 if [[ "${FLAGS_force}" == true ]]; then
-  rm -rf $BUILD_ROOT/build-*
+  rm -rf $OBJ_DIR
+  rm -rf $BIN_DIR
 fi
 
-if [[ ! -n "${FLAGS_reconfigure}" ]] ; then
-  for platform in "${PLATFORMS[@]}"; do
+if [[ "${FLAGS_assert}" == true ]]; then
+  for platform in ${PLATFORMS[@]}; do
```

Comment:
The doc for -A says "Assert that no build dirs exist when running this command." Isn't this loop doing an incomplete check? I would have expected checking that neither $OBJ_DIR or $BIN_DIR exist would be sufficient?

Reply:
I missed that, oops.

Reply:
With the current logic, if `$OBJ_DIR` and `$BIN_DIR` exist but are empty, won't it still pass this assertion? That doesn't seem to be in line with "Assert that no build dirs exist when running this command."?

Reply:
Oh. I see what you mean! The behavior of -A is supposed to be "something has created obj/bin directories so we're going to slam the breaks because someone is touching our stuff". An empty directory should definitely slam the breaks.

Reply:
Sorry, I'm probably not being explicit enough.

Suppose `$OBJ_DIR=foo` and `$BIN_DIR=bar`. If `foo/` and `bar/` both exist but are empty (or indeed are non-empty but include no directories related to , then my understanding is that the check here _won't_ flag that build dirs already exist because it will instead be checking for `foo/${PLATFORMS[@]}`. My expectation is that it should fail.

Additionally, `-e` is probably a slightly better condition to check.

Reply:
Ok, I finally actually found this message. I've addressed this properly now!